evalml/data_checks/outliers_data_check.py

```diff
@@ -56,11 +55,15 @@ def get_IQR(df, k=2.0):
             upper_bound = q3 + (k * iqr)
             return (lower_bound, upper_bound)
 
-        clf = IsolationForest(random_state=self.random_state)
-        clf.fit(X)
-        scores = pd.Series(clf.decision_function(X))
-        lower_bound, upper_bound = get_IQR(scores, k=2)
-        outliers = (scores < lower_bound) | (scores > upper_bound)
-        outliers_indices = outliers[outliers].index.values.tolist()
-        warning_msg = "Row '{}' is likely to have outlier data"
-        return [DataCheckWarning(warning_msg.format(row_index), self.name) for row_index in outliers_indices]
+        lower_bound, upper_bound = get_IQR(X)
+        indices = set()
+        # get the columns that fall out of the bounds, which means they contain outliers
+        for idx, bound in enumerate([lower_bound, upper_bound]):
+            boundary = ((X[bound.keys()] >= bound.values) if idx == 0 else (X[bound.keys()] <= bound.values))[bound.keys()].all()
```

Comment:
I advocate for breaking this up into multiple lines to make it easier to understand how `boundary` is created. Also, I don't think you need to call `[bound.keys()]` everywhere since `X` was already filtered to the columns with numeric data types.